### ISSUE TITLE
xen-api-client.lwt: all calls of the type get_record, get_all_record are failing with Exception: Not_found

### ISSUE THREAD

**Issue by mseri** [xapi-project/xen-api-client#51]
All calls of the type `get_record`, `get_all_record`, are failing with `Exception: Not_found.`

**Comment by gaborigloi** [comment]
The next step is to check whether these calls are failing in `xapi-client` too to narrow down the issue to either the code in this repo or the auto-generated bindings.

**Comment by mseri** [comment]
It is just xen-api-client. We need to make a new release

**Comment by gaborigloi** [comment]
We already have a version tag `v0.9.16` at the HEAD of `master`, and it's already in `xs-opam` 1.8.0.

**Comment by mseri** [comment]
That's the version I was using

**Comment by gaborigloi** [comment]
Since `xapi-client` is not affected, what we could check next is whether the async client is also affected. If not, this was probably introduced by #48 , if it is, then it may have been introduced by the API updates ( #46 ), or the move to PPX ( #43 , #40 ).

**Comment by mseri** [comment]
The calls are broken both in the lwt and async bindings. The async failures are slightly more useful because they print a backtrace. You can actually use the `async_test` executable to see them.

I have tested it using
```ocaml
#require "re.str";;
#require "xen-api-client.async";;
open Async_kernel;;
open Xen_api_async_unix;;
open Xen_api;;
let rpc = make "http://dt21.uk.xensource.com";;
let print_records rs =
  List.iter (fun (n, _) -> Async_unix.Async_print.print_endline n) rs
  |> return
in
Session.login_with_password ~rpc
                            ~uname:"root" ~pwd:"xenroot"
                            ~version:"1.0" ~originator:"pallino"
>>= fun session_id ->
SR.get_all_records ~rpc ~session_id
>>= print_records >>= fun () ->
VBD.get_all_records ~rpc ~session_id
>>= print_records >>= fun () ->
Host.get_all_records ~rpc ~session_id
>>= print_records >>= fun () ->
Console.get_all_records ~rpc ~session_id
>>= print_records >>= fun () ->
VM.get_all_records ~rpc ~session_id
>>= print_records;;

Async_unix.Scheduler.go ~raise_unhandled_exn:false ();;
```
via `utop`. Reordering the `get_all_records` call, or replacing them, you can see the various error (they are not all failing with `Not_found`)

**Comment by gaborigloi** [comment]
Fixed by #55
